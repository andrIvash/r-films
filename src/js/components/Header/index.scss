.header {
  min-height: rem(200px);
  background-image: 
    linear-gradient(to right, rgba(52, 58, 64, 1), rgba(0, 0, 0, 0.75)),
    url('../images/fons2.jpg');
  background-position: center center, center center;
  background-size: contain, contain;
  background-repeat: no-repeat, repeat;

  .container {
    display: flex;
    align-items: flex-start;
    flex-direction: column;
    justify-content: flex-start;
    margin-top: rem(20px);
    text-align: left;
  }

  .header__top {
    width: 100%;
    display: flex;
    justify-content: space-between;
    align-items: center;
    margin-bottom: rem(20px);
  }
}