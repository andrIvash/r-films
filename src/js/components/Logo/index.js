// @flow
import React from 'react';
import { Link } from 'react-router-dom';

const Logo = () => (
  <div className='logo app__logo'>
    <Link to={{ pathname: '/' }}>
      netflixroulette
    </Link>
  </div>
);

export default Logo;
