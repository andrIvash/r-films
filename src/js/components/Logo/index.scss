.logo {
  a {
    font-size: rem(16px);
    font-family: $sign-font;
    font-weight: 700;
    text-decoration: none;
    color: $color-logo; 
    &:hover, &:active {
      color: $color-logo-dark;
    }
  }
}