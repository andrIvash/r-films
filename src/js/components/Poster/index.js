// @flow
import React from 'react';
import type { PosterData } from '../../flow-types.js';

type Props = {
 data: PosterData
}

const Poster = (props: Props) => (
  <div className='poster'>
    <div className='poster__img-wrap'>
      <img alt='file poster' className='poster__img' src={props.data.poster_path} />
    </div>
    <div className='poster__info film'>
      <div className='film__header'>
        <h1 className='film__title'>{props.data.title}</h1>
        <div className='film__score'>{props.data.vote_average}</div>
      </div>
      <h5 className='film__subtitle'>{props.data.tagline}</h5>
      <div className='film__info'>
        <div className='film__year'>{props.data.release_date.substr(0, 4)}</div>
        <div className='film__latency'>{props.data.runtime ? `${props.data.runtime} min` : ''}</div>
      </div>
      <p className='film__desq'>{props.data.overview}
      </p>
    </div>
  </div>
);

export default Poster;
