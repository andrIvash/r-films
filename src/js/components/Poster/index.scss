.poster {
  display: flex;
  width: 100%;
  justify-content: space-between;
  align-items: center;
  flex-wrap: wrap;
  color: $color-layout-dark;
  margin-bottom: rem(20px);
}

.poster__img-wrap {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 30%;
  height: 100%;
  @include tablet() {
    width: 100%;
  }
  .poster__img {
    width: 100%;
    max-width: rem(200px);
    height: auto;
  }
}

.poster__info {
  width: 65%;
  @include tablet() {
    width: 100%;
  }
}

.film__header {
  display: flex;
  justify-content: flex-start;
  align-items: center;
  margin-bottom: rem(10px);
  font-family: $sign-font;

  @include tablet() {
    flex-wrap: wrap;
    flex-direction: column;
    justify-content: center;
  }

  .film__title {
    color: $color-logo;
    width: 80%;
  }

  .film__score {
    display: flex;
    justify-content: center;
    align-items: center;
    width: rem(40px);
    height: rem(40px);
    margin-left: rem(20px);
    border-radius: 50%;
    border: 2px solid $color-layout-dark;
    font-size: rem(14px);  
  }
}

.film__subtitle {
  @include tablet() {
    text-align: center;
  }
}

.film__info {
  display: flex;
  justify-content: flex-start;
  align-items: center;
  margin-bottom: rem(20px);

  @include tablet() {
    flex-wrap: wrap;
    flex-direction: column;
    justify-content: center;
  }
}
.film__year,
.film__latency {
  display: inline-block;
  font-family: $sign-font;
  font-size: rem(13px);
  color: $color-white;
  
  @include tablet() {
    display: inline-block;
    width: 100%;
    margin-left: 0;
  }
}

.film__latency {
  margin-left: rem(20px);

  @include tablet() {
    margin-left: 0;
  }
}

.film__desq {
  margin-bottom: 0;
  font-family: $base-font;
  font-size: rem(16px);
}
