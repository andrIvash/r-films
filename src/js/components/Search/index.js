// @flow
import React, { Component } from 'react';
import { connect } from 'react-redux';
import * as sign from './enterSign.svg';
import FilterSearch from '../FilterSearch';
import SearchButton from '../SearchButton';
import { changeSearchFilter,
  changeSearchText, clearSearchText } from '../../actions';

type Props = {
  submitSearch: (data: string, filter:string) => void,
  searchText: string,
  searchFilter: string,
  clearText: () => void,
  handleChange: (elem: HTMLElement) => void,
  changeFilter: (elem: HTMLElement) => void,
};

export class Search extends Component<Props> {

  handleKeyPress = (event: SyntheticKeyboardEvent<>) => {
    if (event.key === 'Enter') {
      this.handleSearch();
    }
  }

  handleSearch = () => {
    const {searchText, searchFilter, clearText, submitSearch } = this.props;
    if (searchText.length > 0) {
      submitSearch(searchText, searchFilter);
      clearText();
    }
  }

  render() {
    const { searchText, changeFilter, searchFilter } = this.props;
    return (
      <div className='search app__search'>
        <h1 className='search__title'> Find your movie </h1>
        <div className='search__input-wrapper input-group mb-3'>
          <input
            aria-label='Search'
            className='search__input form-control'
            name='search'
            onChange={(ev) => {this.props.handleChange(ev.target.value);}}
            onKeyPress={this.handleKeyPress}
            placeholder='Search phrase...'
            type='text'
            value={searchText}
          />
          <img alt='search sign' className='search__sign' src={sign} />
        </div>
        <div className='search__controls'>
          <FilterSearch
            changeFilter={changeFilter}
            selected={searchFilter}
          />
          <SearchButton handleSearch={this.handleSearch} />
        </div>
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => {
  return {
    changeFilter: (value) => dispatch(changeSearchFilter(value)),
    handleChange: (value) => dispatch(changeSearchText(value)),
    clearText: () => dispatch(clearSearchText()),
  };
};

const mapStateToProps = state => {
  return {
      searchFilter: state.changeSearchFilter.searchFilter,
      searchText: state.changeSearchText.searchText,
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Search);
