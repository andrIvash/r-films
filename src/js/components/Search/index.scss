.search {
  width: 100%;
}

.search__input-wrapper {
  position: relative;

  &:after {
    content: '';
    display: block;
    position: absolute;
    bottom: 0;
    left: 0;
    width: 100%;
    height: 2px;
    background-color: $color-logo;
  }
}

.search__input {
  background-color: $color-dark;
  border-color: $color-dark;
  padding-right: rem(40px);
  position: relative;
  font-family: 'Lato';
  font-weight: 400;


  &:focus {
    outline: none;
    background-color: $color-background;
    border-color: $color-background;
    color: $color-layout;
  }
}

.search__title {
  font-family: 'Lato';
  font-weight: 700;
  color: $color-layout;
  font-size: rem(18px);
  margin-bottom: rem(20px);
  text-transform: uppercase;
}


.search__sign {
  display: block;
  position: absolute;
  top: 50%;
  transform: translateY(-50%);
  right: 0;
  width: rem(30px);
  height: rem(20px);

  img {
    width: rem(30px);
    height: rem(15px);
  }
}

.search__controls {
  display: flex;
  width: 100%;
  justify-content: space-between;
  align-items: center;
  margin-bottom: rem(20px);
}

.search__submit {
  width: rem(100px);
  height: rem(40px);
  font-family: $sign-font;
  font-size: rem(14px);
  margin: 0 rem(5px);
  text-transform: uppercase;
}

.search__submit {
  background-color: $color-logo;
  border-color: $color-logo;
  &:hover {
    background-color: $color-logo-dark;
    border-color: $color-logo-dark;
  }
}