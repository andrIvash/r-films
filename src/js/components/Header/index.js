// @flow
import React, { Component } from 'react';
import Logo from '../Logo';
import Search from '../Search';
import Poster from '../Poster';
import SearchButton from '../SearchButton';
import helpers from '../../helpers';

type Props = {
  onSearch: (data: string, filter:string) => void,
  toSearch: () => void,
  view: string,
  posterData: {
    title: string,
  },
};

const topHeader = (props: Props) => {
  const { view, toSearch } = props;
  if ( view === helpers.views.POSTER ) {
    return (
      <div className='header__top'>
        <Logo />
        <SearchButton handleSearch={toSearch} />
      </div>
    );
  }
  return <div className='header__top'> <Logo /> </div>;
};

const contentHeader = ( props: Props ) => {
  const { view, onSearch, posterData } = props;
  return view === helpers.views.POSTER ?
    <Poster data={posterData} /> :
    <Search submitSearch={onSearch} />;
};

const Header = ( props: Props ) => (
  <div className='header app__header bg-dark'>
    <div className='container'>
      {topHeader(props)}
      {contentHeader(props)}
    </div>
  </div>
);

export default Header;
