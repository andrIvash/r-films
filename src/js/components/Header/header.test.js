import React from 'react';
import renderer from 'react-test-renderer';
import Header from './index';
import Logo from '../Logo';
import Search from '../Search';
import Poster from '../Poster';
import SearchButton from '../SearchButton';

let props = {};

jest.mock('react-router-dom/NavLink', () => 'nav-link');
jest.mock('react-router-dom/Link', () => 'router-link');


jest.mock('../Logo', () => 'logo');
jest.mock('../Search', () => 'search');
jest.mock('../Poster', () => 'poster');
jest.mock('../SearchButton', () => 'search-button');


describe('Header', () => {
  beforeEach(() => {
    props = {
      view: 'POSTER',
      posterData: {
        poster_path: 'http://test',
        release_date: '1924',
        title: 'title',
        genres: ['drama'],
      },
    };
  });
  it('should be defined', () => {
    expect(Header).toBeDefined();
  });
  it('should render correctly', () => {
    const wrapper = renderer.create(<Header {...props} />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
  it('should render right values in POSTER view', () => {

    const wrapper = shallow(
      <Header {...props} /> );

    expect(wrapper.find(Logo)).toHaveLength(1);
    expect(wrapper.find(SearchButton)).toHaveLength(1);
    expect(wrapper.find(Poster)).toHaveLength(1);
  });
  it('should render right values in COMMON view', () => {
    props.view = 'COMMON';

    const wrapper = shallow(
      <Header {...props} /> );

    expect(wrapper.find(Logo)).toHaveLength(1);
    expect(wrapper.find(Search)).toHaveLength(1);
    expect(wrapper.find(Poster)).toHaveLength(0);
  });
});
