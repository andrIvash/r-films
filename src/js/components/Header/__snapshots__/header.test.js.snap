// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly 1`] = `
<div
  className="header app__header bg-dark"
>
  <div
    className="container"
  >
    <div
      className="header__top"
    >
       
      <Logo />
       
    </div>
    <Search />
  </div>
</div>
`;
